src/odemis/odemisd/mdupdater.py

```diff
@@ -352,6 +352,11 @@ def observeFilter(self, filter, comp_affected):
         def updateOutWLRange(pos, fl=filter, comp_affected=comp_affected):
             wl_out = fl.axes["band"].choices[pos["band"]]
             comp_affected.updateMetadata({model.MD_OUT_WL: wl_out})
+            # apply lateral chromatic correction to align with the reference channel
+            apply_transform = fl.getMetadata().get(model.MD_CHROMATIC_COR, None)
```

Comment:
I'd rather keep this as "light" as possible when there is no such metadata. So I'd suggest to check here, early, if the `apply_transform is None`, and immediately leave if so, without reading `.position`. Each of these are a remote function call, so quite slow.